kernel/uGraph.mli

```diff
@@ -16,6 +16,15 @@ type t
 val set_cumulative_sprop : bool -> t -> t
 (** Makes the system incomplete. *)
 
+val set_type_in_type : bool -> t -> t
+
+(** When [type_in_type], functions adding constraints do not fail and
+   may instead ignore inconsistent constraints.
```

Comment:
I'm not sure if this is what we want to do, we could instead shortcut and never add constraints in type-in-type mode.

Reply:
FTR, both behaviours make sense. We would need a spec of type-in-type to decide which one is best suited...